### PR TITLE
br-stream: add restore ts to filter data out of range.

### PR DESCRIPTION
**Pull request by 3pointer**
### What is changed and how it works?

Issue Number: along with https://github.com/pingcap/tidb/pull/32250.
What's Changed:

```
1. add restore ts to filter data out of range.
2. fix the typo `resloved_ts` to `resolved_ts`.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below) 
    see #https://github.com/pingcap/tidb/pull/32250
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with None.
If this PR will be picked to release branch, then a release note is probably required.
```

### CODE REVIEW

File: components/sst_importer/src/sst_importer.rs [hunk 1]
```diff
@@ -342,11 +340,20 @@ impl SSTImporter {
...
+                // we can easily filter the remain keys.
+                break
+            }
+
             smallest_key = smallest_key.map_or_else(
                 || Some(iter_key.clone()),
                 |v: Vec<u8>| Some(v.min(iter_key.clone())),
             );
 
+
```
Comment (YuJuncen) [inline components/sst_importer/src/sst_importer.rs]:
```suggestion
```

File: components/sst_importer/src/sst_importer.rs [hunk 2]
```diff
@@ -342,11 +340,20 @@ impl SSTImporter {
             event_iter.next()?;
             let iter_key = event_iter.key().to_vec();
 
+            let ts = Key::from_encoded(iter_key.clone()).decode_ts()?;
```
Comment (YuJuncen) [inline components/sst_importer/src/sst_importer.rs]: Maybe `Key::decode_ts_from(iter_key.as_slice())`?

```suggestion
            let ts = Key::decode_ts_from(&iter_key)?;
```

**Review by YuJuncen — APPROVED** [review]
rest lgtm

**Comment by kennytm** [comment]
lgtm (after yu juncen's fixes)

**Pull request merged**